### PR TITLE
Added property editor dialog as context menu item for Widgets

### LINKED ISSUE #212: Property sheet for widgets should be on the context menu
When double-clicking on a widget (a somewhat non intuitive operation) the widget area is replaced by it's property sheet. If the widget is small, the property sheet is unusable requiring the widget to be resized, then the property sheet edited, then double-click again, then the, now larger widget needs to be reduced to it's original size.

Instead, the property sheets should just be dialog boxes that pop up on a "Properties..." context menu entry. Then no resizing is required, and users don't have to learn the double-click gesture.

### PR DESCRIPTION
**Pull request by Octogonapus**
Closes #212

image

image

### CODE REVIEW

File: app/src/main/java/edu/wpi/first/shuffleboard/app/components/WidgetTile.java [hunk 1]
```diff
@@ -54,35 +51,18 @@ private WidgetTile() {
...
-    if (newV != null) {
-      newV.addEventHandler(MouseEvent.MOUSE_CLICKED, event -> {
-        if (event.getClickCount() == 2) {
-          toggleShowWidget();
-        }
-      });
-    }
-  }
-
   private Node createCenter() {
```
Comment (SamCarlberg) [inline app/src/main/java/edu/wpi/first/shuffleboard/app/components/WidgetTile.java]: This method can be removed now that the tile can only contain the widget

**Comment by Octogonapus** [comment]
This probably closes #175

File: app/src/main/java/edu/wpi/first/shuffleboard/app/components/WidgetTile.java [hunk 2]
```diff
@@ -100,5 +80,4 @@ public void setShowWidget(boolean showWidget) {
   public void toggleShowWidget() {
```
Comment (SamCarlberg) [inline app/src/main/java/edu/wpi/first/shuffleboard/app/components/WidgetTile.java]: This method (and everything else related to the `showWidget` property) can be removed now that the tile can only contain the widget

**Comment by SamCarlberg** [comment]
This doesn't close #175 since the property sheets are used elsewhere

**Comment by bradamiller** [comment]
It pops up an empty dialog box if there are no properties, for example, with a text field. Also, there is no title on the dialogBox. Should at least have something static like "Edit widget properties". Otherwise seems to work.

It changes the widget properties as you type. So if you change a sliders limits, you see it changing as you are typing. I wonder if sometime (not now, there are more important things to do first), it should have ok, cancel, and apply buttons. Then it might save the properties at the start and put them back on cancel.

**Pull request merged**